modules/social_features/social_profile/modules/social_profile_privacy/src/Service/SocialProfilePrivacyBatchHelper.php

```diff
@@ -0,0 +1,258 @@
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function initOperation(array $args, array &$context) {
+    // Init variables.
+    $limit = (int) $args['limit'];
+    $offset = (int) $context['sandbox'] ?? 0;
+
+    $userStorage = \Drupal::entityTypeManager()->getStorage('user');
+
+    $uids = $userStorage->getQuery()
+      ->condition('status', '1')
+      ->execute();
+    /** @var array $results */
+    $results = $userStorage->loadMultiple($uids);
+
+    // Define total on first call.
+    if (!isset($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = count($results);
+    }
+
+    // Setup results based on retrieved objects.
+    $context['results'] = array_reduce($results,
+      function ($carry, $object) {
+        // Map object results extracted from previous query.
+        $carry[$object->id()] = $object;
+        return $carry;
+      }, $context['results']
+    );
+
+    // Redefine offset value.
+    $context['sandbox']['offset'] = $offset + $limit;
+
+    // Set current step as unfinished until offset is greater than total.
+    $context['finished'] = 0;
+    if ($context['sandbox']['offset'] >= $context['sandbox']['total']) {
+      $context['finished'] = 1;
+    }
+
+    // Setup info message to notify about current progress.
+    $context['message'] = t(
+      'Retrieved @consumed of @total available profiles',
+      [
+        '@consumed' => $context['sandbox']['offset'],
+        '@total' => $context['sandbox']['total'],
+      ]
+    );
+  }
+
+  /**
+   * Process operation to update content retrieved from init operation.
+   *
+   * @param array $args
+   *   Arguments.
+   * @param array $context
+   *   An array that may or may not contain placeholder variables.
+   */
+  public static function updateProcess(array $args, array &$context) {
+    // Define total on first call.
+    if (!isset($context['sandbox']['total'])) {
+      $context['sandbox']['total'] = count($context['results']);
+    }
+
+    // Init limit variable.
+    $limit = $args['limit'];
+
+    // Walk-through all results in order to update them.
+    $count = 0;
+
+    /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+    $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+
+    foreach ($context['results'] as $key => $account) {
+      if (!empty($profile_storage)) {
+        $profile = $profile_storage->loadByUser($account, 'profile');
+        if ($profile instanceof ProfileInterface) {
+          // We need just save the profile. The profile name will be updated by
+          // hook "presave".
+          // @see social_profile_privacy_profile_presave()
+          $profile->save();
+        }
+      }
+
+      // Increment count at one.
+      $count++;
+
+      // Remove current result.
+      unset($context['results'][$key]);
+      if ($count >= $limit) {
```

Comment:
do-while or while loops are better fail-safes, but just a suggestion, can be ignored. 😄

Reply:
@navneet0693 Do you suggest like this?
```
while ($count >= $limit) {
  break;
}
```